app/models/resource_action_workflow.rb

```diff
@@ -40,6 +41,15 @@ def submit_request
     result
   end
 
+  def create_service_order_and_checkout(request)
+    service_order = ServiceOrder.create(:state        => 'ordered',
+                                        :user         => @requester,
+                                        :miq_requests => [request],
+                                        :tenant       => request.tenant)
+    request.update_attributes(:service_order => service_order)
```

Comment:
You are associating the order and the request on the `create` call, I do not think this line is needed.